extractor/main_test.go

```diff
@@ -0,0 +1,412 @@
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, msg := range mockMsgsAddPackage {
+		md := metadataFromMsg(msg)
+
+		// Get output dir
+		outputDir := filepath.Join(tempDir, strings.TrimLeft(msg.Package.Path, "gno.land/"))
+
+		// Write dir before writing metadata
+		if dirWriteErr := os.MkdirAll(outputDir, os.ModePerm); dirWriteErr != nil {
+			t.Fatal(dirWriteErr)
+		}
+
+		// Write the metadata
+		err := writePackageMetadata(md, outputDir)
+		require.NoError(t, err)
+
+		// Read file
+		file, err := os.Open(filepath.Join(outputDir, packageMetadataFile))
+		require.NoError(t, err)
+
+		reader := bufio.NewReader(file)
+		var unmarshalledMetadata Metadata
+
+		raw, isPrefix, err := reader.ReadLine()
+		require.NoError(t, err)
+
+		if isPrefix {
+			t.Fatalf("Metadata longer then buffer max size at %s\n", outputDir)
+		}
+
+		err = json.Unmarshal(raw, &unmarshalledMetadata)
+		require.NoError(t, err)
+
+		t.Cleanup(func() {
+			err := os.RemoveAll(tempDir)
+			if err != nil {
+				fmt.Printf("could not clean up temp dir, %v", err)
+				return
+			}
+		})
+
+		require.Equal(t, md, unmarshalledMetadata)
+	}
+}
+
+func TestWritePackageFiles(t *testing.T) {
+	t.Parallel()
+
+	_, mockMsgsAddPackage := generateMockMsgs(t)
+
+	tempDir, err := os.MkdirTemp(".", "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, msg := range mockMsgsAddPackage {
+		// Get output dir
+		outputDir := filepath.Join(tempDir, strings.TrimLeft(msg.Package.Path, "gno.land/"))
+
+		// Write dir before writing metadata
+		if dirWriteErr := os.MkdirAll(outputDir, os.ModePerm); dirWriteErr != nil {
+			t.Fatal(dirWriteErr)
+		}
+
+		// Write the metadata
+		err := writePackageFiles(msg, outputDir)
+		require.NoError(t, err)
+
+		// Read & compare file
+		for _, f := range msg.Package.Files {
+			contents, err := os.ReadFile(filepath.Join(outputDir, f.Name))
+			require.NoError(t, err)
+			require.Equal(t, f.Body, string(contents))
+		}
+	}
+	t.Cleanup(func() {
+		err := os.RemoveAll(tempDir)
+		if err != nil {
+			fmt.Printf("could not clean up temp dir, %v", err)
+			return
+		}
+	})
+}
+
+// Helpers
+func generateSourceFiles(t *testing.T, mockMsgs []std.Msg) []string {
+	t.Helper()
+
+	tempDir, err := os.MkdirTemp(".", "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var (
+		mockTx    = make([]std.Tx, numTx)
+		testFiles = make([]string, numSourceFiles)
+	)
+
+	// Generate transactions to wrap messages
+	for i := range mockTx { // num
+		mockTx[i] = std.Tx{
+			Msgs: mockMsgs[:msgPerTx],
+		}
+		mockMsgs = mockMsgs[msgPerTx:]
+	}
+
+	// Generate source file names
+	for i := 0; i < numSourceFiles; i++ {
+		testFiles[i] = "sourceFile" + strconv.Itoa(i) + sourceFileType
+	}
+
+	// Generate source files
+	for _, file := range testFiles {
+		filePath := filepath.Join(tempDir, file)
+		if err := os.MkdirAll(filepath.Dir(filePath), os.ModePerm); err != nil {
+			t.Fatal(err)
+		}
+		file, err := os.Create(filePath)
+
+		if err != nil {
+			t.Fatal(err)
+		}
```

Comment:
These created files are never removed, you never propagate out the cleanup method

Reply:
Ah, you're right, you create this subfolder as part of `tempdir`. Please see https://github.com/gnolang/tx-exports/pull/5#discussion_r1427808467, as this create is still at risk of not being cleaned up

Reply:
I've fixed this in the last commit - see [comment](https://github.com/gnolang/tx-exports/pull/5#discussion_r1438300888).